graphblas_algorithms/algorithms/core.py

```diff
@@ -0,0 +1,46 @@
+from graphblas import Matrix, select, agg
+from graphblas.semiring import plus_pair
+
+from graphblas_algorithms.classes.graph import to_undirected_graph, Graph
+from graphblas_algorithms.utils import get_all, not_implemented_for
+
+
+def k_truss_core(G, k):
+    id_to_key = G.id_to_key
+    # Ignore self-edges
+    S = G.get_property("offdiag")
+
+    if k < 3:
+        # Most implementations consider k < 3 invalid,
+        # but networkx leaves the graph unchanged
+        C = S
+    else:
+        # Remove edges not in k-truss
+        nvals_last = S.nvals
+        C = Matrix("int32", S.nrows, S.ncols)
```

Comment:
I think it would be nice to have a helper function that chooses an integer dtype based on a max value.  Here, we could choose the integer dtype based on `S.nrows - 2` as the max possible value.  Actually, I think we can get even fancier and have `max_triangles = min(S.nrows - 2, (S.nvals - 2) // 4)`, which may help with hypersparse graphs.

Also, maybe explicitly use `plus_pair[int_dtype]`.

Reply:
I added a TODO to look into this, but for now, let's just stick with int32.